src/disco/components/Addon.js

```diff
@@ -1,34 +1,66 @@
 import React, { PropTypes } from 'react';
 import classNames from 'classnames';
+import themeAction from 'disco/themePreview';
+import { gettext as _ } from 'core/utils';
 
 import InstallButton from './InstallButton';
+import {
+  validAddonTypes,
+  EXTENSION_TYPE,
+  THEME_TYPE,
+  THEME_PREVIEW,
+  THEME_RESET_PREVIEW,
+} from 'disco/constants';
 
 import 'disco/css/Addon.scss';
 
 
 export default class Addon extends React.Component {
   static propTypes = {
-    id: PropTypes.number.isRequired,
-    type: PropTypes.string.isRequired,
+    accentcolor: PropTypes.string,
```

Comment:
This is the key expected by the theme preview - so I kept it simple rather than making the component have a different key in this case.